### DISCUSSION TITLE
Solving exception bug

### DISCUSSION THREAD

**Discussion by atari-monk** [bilal-fazlani/commanddotnet#430] [General]
Hi @drewburlingame
I have annoying exception
`
CommandDotNet.InvalidConfigurationException: No viable constructors found for Log.Modern.MDI.ConsoleApp.AppProg. Constructor can only contain parameters of type: CommandDotNet.CommandContext,CommandDotNet.IConsole,CommandDotNet.IEnvironment,CommandDotNet.Repl.ReplSession,System.Threading.CancellationToken
`
CommandDotNet entry point class witch appRunner takes as generic type throws this.
This happens in app in witch MicrosoftDependencyInjection is used.
In apps where i use Unity this doesnt pop up.
Do you maybe see what is happening and can give some suggestion ?

**Comment by drewburlingame** [comment]
If it's falling back to CommandDotNet to initialize the instance, then the type wasn't initialized by the DI container.  I suspect the type is not registered with the container.

**Reply by atari-monk** [reply]
They seem to be
![image](https://user-images.githubusercontent.com/5488398/160662983-d01dc8c1-c029-4c63-a211-6b629aed95ba.png)
I dont see what is worng.
I will try to get repl branch and debug with it, see whats what.

**Reply by drewburlingame** [reply]
you don't need the REPL branch to debug if you use https://commanddotnet.bilal-fazlani.com/diagnostics/debug-directive/

https://github.com/bilal-fazlani/commanddotnet/blob/master/CommandDotNet/Execution/ResolverService.cs#L28 is the method to resolve the command class.

**Reply by atari-monk** [reply]
It is a hobby. Enything can create horror.
Problem was removing reference to container.
Thing is i have many helpers and MicrosoftDependencyInjection has this two stage thing, collection and provider.
I ensured that container on AppRunner is set once and last during registration process.
So now it holds proper ref.
Anyway sorry to bother you and thank you for writing back, it helped push the problem.
